### PR TITLE
SDK-139 Create test coverage for the class UsersClient

### PR DESCRIPTION
**Pull request by Haiddi-Avencod**
### Ticket
[SDK-139](https://perzoinc.atlassian.net/browse/SDK-139?atlOrigin=eyJpIjoiM2I5YzVmZDYxYmJjNDExNTg4OWNjMGE5NTBlNzkxNzAiLCJwIjoiaiJ9)

### Description
The class UsersClient was also modified in this ticket because there was a mistake in the methode getSessionUser(). A SymClientException was thrown instead of UnauthorizedException.

### CODE REVIEW

File: symphony-bdk-legacy/symphony-api-client-java/src/test/java/it/clients/symphony/api/UsersClientTest.java [hunk 1]
```diff
@@ -2,96 +2,1029 @@
...
+  private JsonNode stringToJson(String content) throws IOException {
+    final ObjectMapper mapper = new ObjectMapper();
+    final JsonFactory factory = mapper.getFactory();
+    assertNotNull(factory);
+    final JsonParser jsonParser = factory.createParser(content);
+    assertNotNull(jsonParser);
+    final JsonNode jsonNode = mapper.readTree(jsonParser);
+    assertNotNull(jsonNode);
+    return jsonNode;
+  }
```
Comment (symphony-thibault) [inline symphony-bdk-legacy/symphony-api-client-java/src/test/java/it/clients/symphony/api/UsersClientTest.java:361]: You could simplify this method, as you don't necessary need to assert the Jackson API here: 
```java
  private static JsonNode stringToJson(String content) throws IOException {
    return new JsonMapper().readTree(content);
  }
```
Also note that this method can be `static`

File: symphony-bdk-legacy/symphony-api-client-java/src/test/java/it/clients/symphony/api/UsersClientTest.java [hunk 2]
```diff
@@ -2,96 +2,1029 @@
...
+        .withQueryParam("email", equalTo("test_1@symphony.com"))
+        .withQueryParam("local", equalTo("true"))
+        .willReturn(aResponse()
+            .withStatus(204)
+            .withHeader(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_JSON)
+            .withBody("{}")));
+
+    assertNotNull(usersClient);
+
+    final UserInfo userInfo1 = usersClient.getUserFromEmail("test_1@symphony.com", true);
```
Comment (symphony-thibault) [inline symphony-bdk-legacy/symphony-api-client-java/src/test/java/it/clients/symphony/api/UsersClientTest.java:376]: Assignment `userInfo1` is not necessary

File: symphony-bdk-legacy/symphony-api-client-java/src/test/java/it/clients/symphony/api/UsersClientTest.java [hunk 3]
```diff
@@ -2,96 +2,1029 @@
...
-            .withStatus(200)
+            .withStatus(204)
             .withHeader(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_JSON)
-            .withBody(readResourceContent("/response_content/user_client/get_users.json"))));
+            .withBody("{}")));
+    try {
+
+      assertNotNull(usersClient);
+
+      final List<String> emails = Stream.of("test_1@symphony.com", "test_2@symphony.com").collect(Collectors.toList());
```
Comment (symphony-thibault) [inline symphony-bdk-legacy/symphony-api-client-java/src/test/java/it/clients/symphony/api/UsersClientTest.java:909]: Or `final List<String> emails = Arrays.asList("test_1@symphony.com", "test_2@symphony.com");`

**Comment by Haiddi-Avencod** [comment]
Do you want me to fix these things and re-push ?

**Comment by symphony-thibault** [comment]
> Do you want me to fix these things and re-push ?

Yes please @Haiddi-Avencod

**Pull request closed without merging**